File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> answer;
        inorder(root,answer);
        return answer;
    }

    void inorder(TreeNode* rootRecieved, vector<int> &answer){
        if (rootRecieved == nullptr) return;
        inorder(rootRecieved->left,answer);
        answer.push_back(rootRecieved->val);
        inorder(rootRecieved->right,answer);
    }
};

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        array = []
        self.traverse(root,array)
        return array

    def traverse(self,root,array):
        if root == None:
            return
        self.traverse(root.left,array)
        array.append(root.val)
        self.traverse(root.right,array)



    

        